File: Help247.Common/Pagination/PaginationModel.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Help247.Common.Pagination
{
    public class PaginationModel<T> : PaginationBase where T : class
    {
        public int TotalRecords { get; set; }

        public IEnumerable<T> Details { get; set; }
        public dynamic ExtensionData { get; set; }
    }
}

File: Help247/ClientApp/src/app/shared/models/paginationModel.ts
import { PaginationBase } from './pagination-base.model';

export class PaginationModel<T> extends PaginationBase {
  totalRecords: number;
  details: Array<T>;
  extensionData: any;


  constructor(take: number) {
    super();
    this.details = new Array<T>();
    this.skip = 0;
    this.take = take;
    this.totalRecords = 0;
  }
}